### PR TITLE
Add a plan for ruby 2.7

### PR DESCRIPTION
**Pull request by TheLunaticScripter**
Signed-off-by: John Snow <thelunaticscripter@outlook.com>

As part of the release for Chef 16, we need to upgrade to the latest version of ruby. This PR adds a new habitat package for ruby 2.7 following a similar way the Linux plan is in core plans. I think this is a temporary plan that will allow for testing in both the Chef and InSpec projects but at some point, we should just change the version of the main package to 2.7 and then pin the version of the package in Chef and InSpec to 2.6. I'm not very familiar with the ruby release cycle so one of the main questions I have with this method is do we anticipate that ruby 2.6 will still receive some level of security patches and we, therefore, will need to build 2.6 and 2.7 at the same time? This PR will unblock Chef 16 testing but I think a discussion needs to happen about this plan structure both in this repo and in the core-plans.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: bin/ci/verify_a_plan.ps1 [hunk 1]
```diff
@@ -13,8 +13,8 @@ $env:HAB_ORIGIN = 'ci'
 
 Write-Host "--- :8ball: :windows: Verifying $Plan"
 
-$hab_minor_version = (hab --version).split('.')[1]
-if ( -not $? -Or $hab_minor_version -lt 85 ) {
+[Version]$hab_version = (hab --version).split(" ")[1].split("/")[0]
+if ($hab_version -lt [Version]"0.85.0" ) {
```
Comment (robbkidd) [inline bin/ci/verify_a_plan.ps1:17]: `[Version]` Oh, dip! I learned a thing! ✨

File: ruby-plus-devkit27/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Habitat package: ruby-plus-devkit
```
Comment (robbkidd) [inline ruby-plus-devkit27/README.md]:
```suggestion
# Habitat package: ruby-plus-devkit27
```

_shakes fist at version numbers in names_

File: ruby-plus-devkit27/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+Binary package
+
+## Usage
+
+This package provides binaries and libraries specifically for consumption by Ruby-based projects from Chef Software. No support should be expected for direct use of this package by other projects.
+
+For Chef projects on Windows:
+
+* Add this to your package's runtime dependencies. `$pkg_deps=@("chef/ruby-plus-devkit")`
```
Comment (robbkidd) [inline ruby-plus-devkit27/README.md]:
```suggestion
* Add this to your package's runtime dependencies. `$pkg_deps=@("chef/ruby-plus-devkit27")`
```

_shakes fist at version numbers in names_

**Review by robbkidd — CHANGES_REQUESTED** [review]
Apparently I started a review I never submitted. A couple docs/comments corrections for the package name.

File: bin/ci/verify_a_plan.ps1 (hunk 1 above)
Comment (TheLunaticScripter) [reply bin/ci/verify_a_plan.ps1:17]: .net_power

**Comment by robbkidd** [comment]
Ugh. My suggested edits did not include sign-offs. 

@TheLunaticScripter You wanna rebase (either squashing or rewriting) and force push or would you like me to?

**Review by robbkidd — APPROVED** [review]

File: ruby-plus-devkit27/plan.ps1 [hunk 4]
```diff
@@ -0,0 +1,51 @@
+$pkg_name="ruby-plus-devkit27"
```
Comment (robbkidd) [inline ruby-plus-devkit27/plan.ps1]: @TheLunaticScripter Looking at this again, should the package name be `ruby27-plus-devkit` maybe? 🤦

**Review by robbkidd — CHANGES_REQUESTED** [review]
Yea. I recommend the package be named `ruby27-plus-devkit`. 😬

File: ruby-plus-devkit27/plan.ps1 (hunk 4 above)
Comment (TheLunaticScripter) [reply ruby-plus-devkit27/plan.ps1]: Yeah that makes sense since it's the ruby that's 2.7 not the devkit part. I'll change that.

**Review by btm — APPROVED** [review]

**Review by robbkidd — APPROVED** [review]

**Pull request merged**